src/pages/resultsView/download/DownloadTab.tsx

```diff
@@ -434,4 +482,32 @@ export default class DownloadTab extends React.Component<IDownloadTabProps, {}>
     {
         onMobxPromise(this.transposedCnaDataText, text=>fileDownload(text, "cna_transposed.txt"));
     }
+
+    @action
+    private handleQueryButtonClick(querySampleIds: string) {
+        this.props.store.selectedSampleIdsForQueryButton = querySampleIds;
+        this.props.store.queryFormVisible = true;
+    }
+
+    @action
+    private async handleVisualStudyButtonClick(alterations: ICaseAlteration[]) {
+        let studies = _.reduce(_.groupBy(alterations, "studyId"), (acc: { id: string; samples: string[] }[], alteration, studyId) => {
+            acc.push({
+                id: studyId,
+                samples: alterations.map(alteration => alteration.sampleId)
+            })
+            return acc;
+        }, []);
+
+        let parameters = {
+            name: "visual study",
```

Comment:
@jjgao if we're going to create a virtual study ("visual study" is an error), what should it be named?  Maybe we can grab a name from the lable we are using for this row in table @dippindots